### PR TITLE
SwiftPM: Test a corner case

### PR DESCRIPTION
**Pull request by fviernau**
See individual commits.

### CODE REVIEW

File: plugins/package-managers/swiftpm/src/funTest/assets/projects/synthetic/project-without-lockfile-and-deps/Package.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    ],
+    products: [
+        .library(
+            name: "MyPackage",
+            targets: ["MyPackage"]),
+    ],
+    targets: [
+        .target(name: "MyPackage")
+    ]
+)
```
Comment (sschuberth) [inline plugins/package-managers/swiftpm/src/funTest/assets/projects/synthetic/project-without-lockfile-and-deps/Package.swift:19]: Nit: Missing final newline.

File: plugins/package-managers/swiftpm/src/funTest/kotlin/SwiftPmFunTest.kt [hunk 2]
```diff
@@ -88,4 +88,20 @@ class SwiftPmFunTest : WordSpec({
             result.withInvariantIssues().toYaml() should matchExpectedResult(expectedResultFile, definitionFile)
         }
     }
+
+    "Analyzing a definition file without a lockfile without dependencies" should {
```
Comment (sschuberth) [inline plugins/package-managers/swiftpm/src/funTest/kotlin/SwiftPmFunTest.kt:90]: Isn't "without a lockfile" implied by "without dependencies"? If you want to mention both, maybe say "with no dependencies nor lockfile"?

File: plugins/package-managers/swiftpm/src/funTest/assets/projects/synthetic/expected-output-project-without-deps.yml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  vcs:
+    type: ""
+    url: ""
+    revision: ""
+    path: ""
+  vcs_processed:
+    type: "Git"
+    url: "<REPLACE_URL_PROCESSED>"
+    revision: "<REPLACE_REVISION>"
+    path: "plugins/package-managers/swiftpm/src/funTest/assets/projects/synthetic/project-without-deps"
```
Comment (sschuberth) [inline plugins/package-managers/swiftpm/src/funTest/assets/projects/synthetic/expected-output-project-without-deps.yml:16]: Should this stay at `"<REPLACE_PATH>"`?

**Review by sschuberth — APPROVED** [review]

**Comment by fviernau** [comment]
Merged despite failing unrelated PipenvFunTest.
